### PR TITLE
Matched effects funcs

### PR DESCRIPTION
**Pull request by savaughn**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 10 functions (+0.07%) on `us`.

⚠️ This PR introduces a warning:
- src/effects/ring_blast.c:24: warning: assignment from incompatible pointer type

File: include/effects.h [hunk 1]
```diff
@@ -1121,8 +1128,16 @@ typedef struct ConfettiFXData {
 } ConfettiFXData; // size = 0x34
 
 typedef struct SnowfallFXData {
-    /* 0x00 */ char todo[0];
-} SnowfallFXData; // size = unknown
+    s32 unk_00;
```
Comment (ethteck) [inline include/effects.h:1131]: struct comments please

File: src/effects/ring_blast.c [hunk 2]
```diff
@@ -2,13 +2,67 @@
...
+    data->pos.x = posX;
+    data->pos.y = posY;
+    data->pos.z = posZ;
+    data->unk_10 = arg4;
+    data->unk_24 = 0;
+    if (arg0 == 0) {
+        data->unk_28 = 0;
+        return;
+    }
+    data->unk_28 = 0x28;
```
Comment (ethteck) [inline src/effects/ring_blast.c:43]: decimal plez

File: src/effects/ring_blast.c [hunk 3]
```diff
@@ -2,13 +2,67 @@
...
+    data->unk_20 = data->lifeTime * 8.0f / data->unk_1C;
+
+    data->lifeTime++;
+    data->timeLeft--;
+
+    if (data->timeLeft < 0) {
+        shim_remove_effect(effect);
+        return;
+    }
+    data->unk_24 = data->unk_24 + data->unk_28;
```
Comment (ethteck) [inline src/effects/ring_blast.c:61]:
```suggestion
    data->unk_24 += data->unk_28;
```

File: src/effects/ring_blast.c [hunk 4]
```diff
@@ -2,13 +2,67 @@
...
+    if (data->unk_00 == 0) {
+        data->unk_10 = data->unk_10 + 0.1;
```
Comment (ethteck) [inline src/effects/ring_blast.c:63]:
```suggestion
        data->unk_10 += 0.1;
```

File: src/effects/rising_bubble.c [hunk 5]
```diff
@@ -1,27 +1,60 @@
...
+    data->pos.z = posZ;
+    data->unk_14 = 0;
+    data->lifeTime = 0;
+    if (arg4 != 0.0f) {
+        data->unk_10 = ((shim_rand_int(3) * 0.1) + 0.7) * 0.4;
+    } else {
+        data->unk_10 = 1.f;
+    }
+    data->timeLeft = 0x20;
+    data->unk_20 = shim_rand_int(0x1F);
```
Comment (ethteck) [inline src/effects/rising_bubble.c:38]: decimal plez

File: src/effects/snowfall.c [hunk 6]
```diff
@@ -1,13 +1,54 @@
...
 INCLUDE_ASM(s32, "effects/snowfall", func_E008A000);
 
-INCLUDE_ASM(s32, "effects/snowfall", snowfall_main);
+void func_E008A000(SnowfallFXData* data);
+
+EffectInstance* snowfall_main(s32 arg0, s32 arg1) {
+    EffectBlueprint effectBp;
+    EffectInstance* effect;
+    SnowfallFXData* data;
+    s32 index;
```
Comment (ethteck) [inline src/effects/snowfall.c:18]: name this `i` please, and make it the last var declared

File: src/effects/snowfall.c [hunk 7]
```diff
@@ -1,13 +1,54 @@
...
+    s32 temp_s0;
```
Comment (ethteck) [inline src/effects/snowfall.c:19]: name this one numParts

Comment (ethteck) [reply src/effects/snowfall.c:19]: also can it be initialized when declared like in most effect functions?

File: src/effects/snowfall.c [hunk 8]
```diff
@@ -1,13 +1,54 @@
...
+    effectBp.unk_14 = 0;
+    effectBp.effectID = EFFECT_SNOWFALL;
+
+    effect = shim_create_effect_instance(&effectBp);
+    effect->numParts = temp_s0;
+
+    data = effect->data.snowfall = shim_general_heap_malloc(temp_s0 * sizeof(*data));
+    ASSERT(data != NULL);
+
+    data->timeLeft = 0x64;
```
Comment (ethteck) [inline src/effects/snowfall.c:35]: decimal for this and unk_28 below

File: src/effects/thunderbolt_ring.c [hunk 9]
```diff
@@ -1,14 +1,85 @@
...
+    timeLeft = data->timeLeft;
+    data->lifeTime++;
+    lifeTime = data->lifeTime;
+
+    if (timeLeft < 0) {
+        shim_remove_effect(effect);
+        return;
+    }
+    if (timeLeft < 8) {
+        data->unk_2C = data->unk_2C + ((8 - timeLeft) * 0.3);
```
Comment (ethteck) [inline src/effects/thunderbolt_ring.c:74]:
```suggestion
        data->unk_2C += (8 - timeLeft) * 0.3;
```

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/effects/rising_bubble.c [hunk 10]
```diff
@@ -1,27 +1,60 @@
...
+    data->unk_00 = arg0;
+    data->pos.x = posX;
+    data->pos.y = posY;
+    data->pos.z = posZ;
+    data->unk_14 = 0;
+    data->lifeTime = 0;
+    if (arg4 != 0.0f) {
+        data->unk_10 = ((shim_rand_int(3) * 0.1) + 0.7) * 0.4;
+    } else {
+        data->unk_10 = 1.f;
```
Comment (Brotenko) [inline src/effects/rising_bubble.c:35]: I didn't know gcc just flies with this :D
```suggestion
        data->unk_10 = 1.0f;
```

File: src/effects/rising_bubble.c [hunk 11]
```diff
@@ -1,27 +1,60 @@
...
+    }
+    data->timeLeft = 0x20;
```
Comment (Brotenko) [inline src/effects/rising_bubble.c:37]: To quote the greatest of the greats: "decimal plez"

**Review by Brotenko — CHANGES_REQUESTED** [review]
🙏😔

File: src/effects/rising_bubble.c [hunk 12]
```diff
@@ -1,27 +1,60 @@
...
+    data->unk_20 = shim_rand_int(0x1F);
+    data->unk_24 = posY + arg4;
+}
+void rising_bubble_init(EffectInstance* effect) {
```
Comment (nanaian) [inline src/effects/rising_bubble.c:41]:
```suggestion
}

void rising_bubble_init(EffectInstance* effect) {
```

File: src/effects/thunderbolt_ring.c [hunk 13]
```diff
@@ -1,14 +1,85 @@
...
+    }
+    if (lifeTime < 16) {
+        data->unk_24 = (lifeTime * 16) + 15;
+    }
+    if (timeLeft < 8) {
+        data->unk_24 = (timeLeft << 5) + 31;
```
Comment (nanaian) [inline src/effects/thunderbolt_ring.c:80]: Change shift to multiplication

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/effects/thunderbolt_ring.c (hunk 13 above)
Comment (Brotenko) [reply src/effects/thunderbolt_ring.c:80]: No promises, but this should work:
```suggestion
        data->unk_24 = (timeLeft * 32) + 31;
```

File: src/effects/snowfall.c (hunk 7 above)
Comment (savaughn) [reply src/effects/snowfall.c:19]: yeah, I'm not sure why I did that. I think that this was my first effects main.

File: src/effects/rising_bubble.c (hunk 10 above)
Comment (savaughn) [reply src/effects/rising_bubble.c:35]: lol force of habit

File: include/effects.h [hunk 14]
```diff
@@ -1797,18 +1797,18 @@ typedef struct HieroglyphsFXData {
...
+    /* 0x14 */ f32 scaleY;
+    /* 0x18 */ char unk_18[0x20];
+    /* 0x38 */ s32 unk_38;
+    /* 0x3C */ s32 unk_3C;
+    /* 0x40 */ s32 unk_40;
+    /* 0x44 */ char unk_44[0x4];
+    /* 0x48 */ s32 unk_48;
+    /* 0x4C */ s32 unk_4C;
+    /* 0x50 */ s32 unk_50;
 } MiscParticlesFXData; // size = ?
```
Comment (Brotenko) [inline include/effects.h:1812]:
```suggestion
} MiscParticlesFXData; // size = 0x54
```

**Review by Brotenko — CHANGES_REQUESTED** [review]
One teeny, tiny, last, widdle, bit thing 🤏

File: include/effects.h (hunk 14 above)
Comment (ethteck) [reply include/effects.h:1812]: We don't know the size of this yet, as it's not decompiled

Comment (Brotenko) [reply include/effects.h:1812]: Ignore :)

Comment (savaughn) [reply include/effects.h:1812]: nah I just saw that it was missing comments too

**Pull request merged**